### DISCUSSION TITLE
Error deploying Next.JS With Mongo "Error: querySrv ETIMEOUT _mongodb._tcp."

### DISCUSSION THREAD

**Discussion by jignaciomartinez** [vercel/community#6870] [Help] [answered]
### Summary

Suddenly I started getting this error on deploy, and the build fails.

Error: querySrv ETIMEOUT _mongodb._tcp.<clusterURL>
    at QueryReqWrap.onresolve [as oncomplete] (node:internal/dns/promises:291:17) {
  errno: undefined,
  code: 'ETIMEOUT',
  syscall: 'querySrv',
  hostname: '_mongodb._tcp.<clusterURL>',
  digest: '3631126898'
}
Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
Error: querySrv ETIMEOUT _mongodb._tcp.<clusterURL>
    at QueryReqWrap.onresolve [as oncomplete] (node:internal/dns/promises:291:17)
Error occurred prerendering page "/api/cota". Read more: https://nextjs.org/docs/messages/prerender-error
Error: querySrv ETIMEOUT _mongodb._tcp.<clusterURL>
    at QueryReqWrap.onresolve [as oncomplete] (node:internal/dns/promises:291:17)
    
    Everything works on dev, connecting to the same cluster.

    npm run build runs without any problem on my Mac.

### Steps to Reproduce

Just try to run a deploy on Vercel, and I got the error.

**Comment by TGilany** [comment]
I have the same issue with SvelteKit, seems like an issue from today.
I also tried to redeploy a previous version that worked yesterday and it failed as well. No matter what I try it won't work.

```BASH
Error: querySrv ETIMEOUT _mongodb._tcp.*******.skj2pbu.mongodb.net
17:52:41.708 | at QueryReqWrap.onresolve [as oncomplete] (node:internal/dns/promises:291:17)
17:52:41.708 | Emitted 'error' event on Worker instance at:
17:52:41.708 | at [kOnErrorMessage] (node:internal/worker:326:10)
17:52:41.708 | at [kOnMessage] (node:internal/worker:337:37)
17:52:41.708 | at MessagePort.<anonymous> (node:internal/worker:232:57)
17:52:41.709 | at [nodejs.internal.kHybridDispatch] (node:internal/event_target:825:20)
17:52:41.709 | at MessagePort.<anonymous> (node:internal/per_context/messageport:23:28) {
17:52:41.709 | errno: undefined,
17:52:41.709 | code: 'ETIMEOUT',
17:52:41.709 | syscall: 'querySrv',
17:52:41.709 | hostname: '_mongodb._tcp.*******.skj2pbu.mongodb.net'
17:52:41.709 | }
17:52:41.709 |  
17:52:41.709 | Node.js v20.13.1
17:52:41.762 | ELIFECYCLE  Command failed with exit code 1.
17:52:41.776 | Error: Command "pnpm run build" exited with 1
17:52:42.271
```

**Comment by charcharchu** [comment]
also having exactly the same issue today with nextjs deployment

**Comment by jignaciomartinez** [comment]
I used a workaround by building it using a standard MongoDB URI (`mongo://...`) instead of `mongo+srv://`. You can find it by looking for the `mongodriver2.x` URI. However, I am now unable to retrieve data. I'm not sure if I broke something or if it's an issue related to Vercel.

**Comment by swarnava** [comment]
Hi people,

Swarnava from Vercel Customer Success team here. We have received few reports from customers and we are currently investigating this.

https://www.vercel-status.com/incidents/9twxvphwx48c

**Comment by swarnava** [comment]
We are sharing update here: https://www.vercel-status.com/incidents/9twxvphwx48c

**Reply by TGilany** [reply]
Thanks for the quick response!

**Comment by swarnava** [comment, marked as answer]
There was an issue with Node v20.13.1. We just rolled back to v20.12.1 and it should be fixed now.

**Reply by TGilany** [reply]
Seems like it's working now, thanks!

**Comment by GodsroadFirima** [comment]
Mine is still showing that error
I have node v20.12.0 installed

**Reply by akhoshrozeh** [reply]
Did you try Node v20.13.1?

**Reply by GodsroadFirima** [reply]
No sir

[a comment by an automated account is not shown]
